### PR TITLE
Fix links in documentation

### LINKED ISSUE #351: Generated function documentation has broken links
http://nimi-python.readthedocs.io/en/master/nidmm/attributes.html

For example: "Use niDMM Get Next Interchange Warning to extract interchange warnings." links to dmmviref.chm::/niDMM_Get_Next_Interchange_Warning.html

### LINKED ISSUE #391: Javascript shown in functions.rst that create warnings
Functions.rst shows
<javascript:LaunchMergedHelp('NI_DC_Power_Supplies_Help.chm',%20'NIDCPowerCRef.chm',%20'cviniDCPower_InitializeWithChannels.html')>`__

Which creates warnings

### LINKED ISSUE #427: should we retain links in method documentation ?
There are some links to topic available in session.py for nidcpower. 
```
def configure_aperture_time(self, aperture_time, units):
        '''configure_aperture_time

        Vistatus configure_aperture_time(ViSession vi, ViConstString
        channelName, ViReal64 apertureTime, ViInt32 units);

        Configures the aperture time on the specified channel(s).

        The supported values depend on the **units**. Refer to the *Aperture
        Time* topic for your device in the *NI DC Power Supplies and SMUs Help*
        for more information. In general, devices support discrete
        **apertureTime** values, and if you configure **apertureTime** to some
        unsupported value, NI-DCPower coerces it up to the next supported value.

        Refer to the *Measurement Configuration and Timing* or *DC Noise
        Rejection* topic for your device in the *NI DC Power Supplies and SMUs
        Help* for more information about how to configure your measurements.

        **Related Topics:**

        `Aperture
        Time <http://zone.ni.com/reference/en-XX/help/370384T-01/dmm/aperture/>`__

        Note:
        This function is not supported on all devices. Refer to `Supported
        Functions by
        Device <javascript:LaunchMergedHelp('NI_DC_Power_Supplies_Help.chm',%20'NIDCPowerCRef.chm',%20'supportedFunctions.html')>`__
        for more information about supported devices.

        Args:
            channel_name (string): Specifies the output channel(s) to which this configuration value
                applies. Specify multiple channels by using a channel list or a channel
                range. A channel list is a comma (,) separated sequence of channel names
                (for example, 0,2 specifies channels 0 and 2). A channel range is a
                lower bound channel followed by a hyphen (-) or colon (:) followed by an
                upper bound channel (for example, 0-2 specifies channels 0, 1, and 2).
                In the Running state, multiple output channel configurations are
                performed sequentially based on the order specified in this parameter.
            aperture_time (float): Specifies the aperture time. Refer to the *Aperture Time* topic for your
                device in the *NI DC Power Supplies and SMUs Help* for more information.
            units (int): Specifies the units for **apertureTime**.
                **Defined Values**:

                +----------------------------------------+------------------------------+
                | NIDCPOWER_VAL_SECONDS (1028)           | Specifies seconds.           |
                +----------------------------------------+------------------------------+
                | NIDCPOWER_VAL_POWER_LINE_CYCLES (1029) | Specifies Power Line Cycles. |
                +----------------------------------------+------------------------------+
        '''
```

Above function for nidcpower has following links available. Not sure how user will be able to open links specified to chm files from python file.
```
<javascript:LaunchMergedHelp('NI_DC_Power_Supplies_Help.chm',%20'NIDCPowerCRef.chm',%20'supportedFunctions.html')>`__
`Aperture
        Time <http://zone.ni.com/reference/en-XX/help/370384T-01/dmm/aperture/>`__
```

### PR DESCRIPTION
**Pull request by texasaggie97**
[X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/<reponame>/blob/master/CONTRIBUTING.md).
~~[ ] I've updated [CHANGELOG.md](https://github.com/ni/<reponame>/blob/master/CHANGELOG.md) if applicable.~~
### What does this Pull Request accomplish?
* Fixes the generated links in documentation
  * Updates function.py metadata with fixed function names
    * Found issue that was changing the case and causing links to not work
  * Allow driver specific urls for documenation
    * The link in documentation need to look like this:
      ```
      `Devices Overview <REPLACE_DRIVER_SPECIFIC_URL_1(devices)>`__
      ```
    * Add driver_urls to driver specific config:
      ``` python
      'driver_urls': {
          'REPLACE_DRIVER_SPECIFIC_URL_1': 'http://zone.ni.com/reference/en-XX/help/370384T-01/dmm/{0}/',
      },
      ```
    * Item in parens will be passed to python `format()` function in order. This way the fixed up link will look like:
      ```
      `Devices Overview <http://zone.ni.com/reference/en-XX/help/370384T-01/dmm/devices/>`__
      ```
### List issues fixed by this Pull Request below, if any.
* Fix #427 
* Fix #351
* Fix #391 

### What testing has been done?
* Travis
* System tests
* Visual inspection of documentation

### CODE REVIEW

File: generated/nidmm/session.py [hunk 1]
```diff
@@ -2163,18 +2156,18 @@ def read_multi_point(self, array_size, maximum_time=-1):
...
-            reading_array (list of float): An array of measurement values.
+            reading_array (string): An array of measurement values.
 
                 Note:
                 The size of the **Reading_Array** must be at least the size that you
                 specify for the **Array_Size** parameter.
             actual_number_of_points (int): Indicates the number of measured values actually retrieved from the DMM.
         '''
-        reading_array_ctype = (visatype.ViReal64 * array_size)()
+        reading_array_ctype = (visatype.ViChar * array_size)()
```
Comment (marcoskirsch) [inline generated/nidmm/session.py]: This seems wrong... Is this fixed now in master?

File: generated/nidmm/tests/mock_helper.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def __init__(self):
         self._defaults['Disable']['return'] = 0
         self._defaults['Fetch'] = {}
         self._defaults['Fetch']['return'] = 0
-        self._defaults['Fetch']['reading'] = None
+        self._defaults['Fetch']['Reading'] = None
```
Comment (marcoskirsch) [inline generated/nidmm/tests/mock_helper.py]: Is this intentional?

File: src/nidmm/metadata/functions.py [hunk 3]
```diff
@@ -763,7 +762,7 @@
             {
                 'direction': 'in',
                 'enum': None,
-                'name': 'action',
+                'name': 'Action',
```
Comment (marcoskirsch) [inline src/nidmm/metadata/functions.py]: Is this how NI-DMM's CVI FP file has it? Why did this change?
This probably should be filed as an NI internal bug.

File: generated/nidmm/tests/mock_helper.py (hunk 2 above)
Comment (texasaggie97) [reply generated/nidmm/tests/mock_helper.py]: Yes, this matches the FP file.

File: src/nidmm/metadata/functions.py (hunk 3 above)
Comment (texasaggie97) [reply src/nidmm/metadata/functions.py]: Yes, this matches the FP file. All parameter names use title case.

**Pull request merged**